jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ThreadMXBeanImpl.java

```diff
@@ -255,7 +264,19 @@ public ThreadInfo getThreadInfo(long id) {
 		}
 
 		// Create an array and populate with individual ThreadInfos
-		return this.getMultiThreadInfoImpl(ids, maxDepth, false, false);
+		ThreadInfoBase[] infoBases = this.getMultiThreadInfoImpl(ids, maxDepth, false, false);
+		return wrapBases(infoBases);
+	}
+
+	/**
+	 * Create ThreadInfo objects containing ThreadInfoBase objects.
+	 * @param infoBases containers for the actual thread information
+	 * @return ThreadInfo array whose elements wrap the elements of infoBases 
+	 */
+	private static ThreadInfo[] wrapBases(ThreadInfoBase[] infoBases) {
+		ThreadInfo[] result = Arrays.stream(infoBases)
+				.map(b -> makeThreadInfo(b)).toArray(ThreadInfo[]::new);
```

Comment:
`b -> makeThreadInfo(b)` can be replaced by the method reference `ThreadMXBeanImpl::makeThreadInfo`.

It might make sense to inline the use of `doPrivileged` so its cost is amortized over the the array of object rather than paying the price for each object.

Reply:
Also, the pair of methods should be named similarly, either `wrapBase` and `wrapBases` or `makeThreadInfo` and `makeThreadInfos` rather than a mix.

Reply:
All good ideas.  Done.